hooks/src/internal.d.ts

```diff
@@ -30,14 +32,14 @@ export interface ComponentHooks {
 	_pendingEffects: EffectHookState[];
 }
 
-export interface Component extends PreactComponent<any, any> {
+export interface Component extends Omit<PreactComponent<any, any>, '_renderCallbacks'> {
```

Comment:
This PR adds a healthy sprinkling of `Omit` which I think is correct given our use cases -- generally, our internal types intentionally clash with the base/user-exposed types.